src/redux/auth/auth.actions.ts

```diff
@@ -11,20 +11,20 @@ export const userLogin = createAsyncThunk(
         username,
         password,
       });
-      localStorage.setItem("loggedInUser", JSON.stringify(res.data));
-      console.log(res.data);
-      return res.data;
+      if (res.data) {
+        localStorage.setItem("loggedInUser", JSON.stringify(res.data));
+        return res.data;
+      }
     } catch (error) {
-      console.log(error);
-      throw error;
+      throw new Error("Invalid username or password. Please try again");
     }
   }
 );
 
 export const setLoggedInUser = createAsyncThunk(
```

Comment:
It seems ALL your actions here are returning type any, as I mentioned before, please and types to the generics here...